server/src/main/java/org/elasticsearch/index/codec/vectors/diskbbq/IVFVectorsReader.java

```diff
@@ -43,20 +42,37 @@
 
 import static org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsReader.SIMILARITY_FUNCTIONS;
 import static org.elasticsearch.index.codec.vectors.diskbbq.ES920DiskBBQVectorsFormat.DYNAMIC_VISIT_RATIO;
+import static org.elasticsearch.index.codec.vectors.diskbbq.ES920DiskBBQVectorsFormat.VERSION_DIRECT_IO;
 
 /**
  * Reader for IVF vectors. This reader is used to read the IVF vectors from the index.
  */
 public abstract class IVFVectorsReader extends KnnVectorsReader {
 
+    private record FlatVectorsReaderKey(String formatName, boolean useDirectIO) {
+        private FlatVectorsReaderKey(FieldEntry entry) {
+            this(entry.rawVectorFormatName, entry.useDirectIOReads);
+        }
+
+        @Override
+        public String toString() {
+            return formatName + (useDirectIO ? " with Direct IO" : "");
+        }
+    }
+
     private final IndexInput ivfCentroids, ivfClusters;
     private final SegmentReadState state;
     private final FieldInfos fieldInfos;
     protected final IntObjectHashMap<FieldEntry> fields;
-    private final Map<String, FlatVectorsReader> rawVectorReaders;
+    private final Map<FlatVectorsReaderKey, FlatVectorsReader> rawVectorReaders;
+
+    @FunctionalInterface
+    public interface GetFormatReader {
+        FlatVectorsReader getReader(String formatName, boolean useDirectIO) throws IOException;
+    }
```

Comment:
Do we want to make this a more general interface? Maybe put it in its own class outside of IVF reader?